Python/getargs.c

```diff
@@ -616,6 +624,15 @@ convertsimple(PyObject *arg, const char **p_format, va_list *p_va, int flags,
               char *msgbuf, size_t bufsize, freelist_t *freelist)
 {
 #define RETURN_ERR_OCCURRED return msgbuf
+#define HANDLE_NULLABLE                 \
```

Comment:
I know understand why the implementation is more complex. You essentially need to add that macro after every flag that is parsed instead of handling it as if it were a regular character.